modules/datahandling/datahandler.go

```diff
@@ -25,33 +37,80 @@ func (dh DataHandler) Handle(wsID uint64, messageType int, message []byte) error
 		return err
 	}
 
-	// automatically delegates if the request is authenticated or not
-	fullR, err := getFullRequest(req)
+	// automatically determines if the request is authenticated or not
+	fullRequest, err := getFullRequest(req)
+
+	var closures []dhClosure
 
 	if err != nil {
-		utils.LogOnError(err, "Failed to construct full request")
-		return err
+		res := new(serverMessageWrapper)
+		res.Timestamp = time.Now().UnixNano()
+		res.Type = "Responce"
+		if err == ErrAuthenticationFailed {
+			utils.LogOnError(err, "User not logged in")
+			res.ServerMessage = response{
+				Status: unauthorized,
+				Tag:    req.Tag,
+				Data:   struct{}{}}
+		} else {
+			utils.LogOnError(err, "Failed to construct full request")
+			res.ServerMessage = response{
+				Status: unimplemented,
+				Tag:    req.Tag,
+				Data:   struct{}{}}
+		}
+		closures = []dhClosure{toSenderClosure{msg: res}}
+	} else {
+		closures, err = fullRequest.process(dh.Db)
+		if err != nil {
+			utils.LogOnError(err, "Failed to handle process request")
+		}
+	}
+
+	for _, closure := range closures {
+		erro := closure.call(dh)
+		if erro != nil {
+			utils.LogOnError(erro, "Failed to complete continuation")
+		}
 	}
 
-	response, notification, err := fullR.process()
+	return err
+}
+
+func authenticate(abs abstractRequest) bool {
+	fmt.Println("AUTHENTICATION IS NOT IMPLEMENTED YET")
+	// TODO (non-immediate/required): implement user authentication
+	return true
+}
 
+// SendToSender is the function that will forward a server message back to the client
+func (dh DataHandler) sendToSender(msg *serverMessageWrapper) error {
+	msgJSON, err := json.Marshal(msg)
 	if err != nil {
-		utils.LogOnError(err, "Failed to process request")
 		return err
 	}
-
-	if response != nil {
-		// TODO: send on rabbit
-	}
-
-	if notification != nil {
-		// TODO: send on rabbit
+	dh.MessageChan <- rabbitmq.AMQPMessage{
+		Headers:     make(map[string]interface{}),
+		RoutingKey:  strconv.FormatUint(dh.WebsocketID, 10),
+		ContentType: msg.Type,
+		Persistent:  false,
+		Message:     msgJSON,
 	}
-
 	return nil
 }
 
-func authenticate(abs abstractRequest) bool {
-	// TODO: implement this
-	return true
+// SendToChannel is the function that will forward a server message to a channel based on the given routing key
+func (dh DataHandler) sendToChannel(msg *serverMessageWrapper) error {
```

Comment:
Perhaps rename this to sendToRabbitMQ? Channel might be a little ambiguous in this context.